elide-async/src/main/java/com/yahoo/elide/async/service/AsyncQueryDAO.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020, Yahoo Inc.
+ * Licensed under the Apache License, Version 2.0
+ * See LICENSE file in project root for terms.
+ */
+package com.yahoo.elide.async.service;
+
+import java.util.UUID;
+
+import com.yahoo.elide.Elide;
+import com.yahoo.elide.async.models.AsyncQuery;
+import com.yahoo.elide.async.models.AsyncQueryResult;
+import com.yahoo.elide.core.DataStore;
+
+/**
+ * Utility interface which uses the elide datastore to modify, update and create
+ * AsyncQuery and AsyncQueryResult Objects
+ */
+public interface AsyncQueryDAO {
+
+    /**
+     * Set elide object
+     * @param elide Elide Object.
+     */
+    public void setElide(Elide elide);
+
+    /**
+     * Set data store object
+     * @param dataStore Datastore Object from Elide.
+     */
+    public void setDataStore(DataStore dataStore);
+
+    /**
+     * This method updates the model for AsyncQuery with passed value.
+     * @param asyncQueryId Unique UUID for the AsyncQuery Object
+     * @param updateFunction Functional interface for updating AsyncQuery Object
+     * @return AsyncQuery Object
+     */
+    public AsyncQuery updateAsyncQuery(UUID asyncQueryId, UpdateQuery updateFunction);
```

Comment:
I don't think we should expose implementation details like UpdateQuery in a DAO interface.  

We should have methods like ```updateStatus(UUID asyncQueryId, QueryStatus status)```